File: WarehouseManagementWeb.Application/Dto/Input/ResourceShipment/UpdateResourceShipmentInput.cs
using WarehouseManagementWeb.Domain.Enums;

namespace WarehouseManagementWeb.Application.Dto.Input.ResourceShipment
{
    /// <summary>
    /// Класс входной модели редактирования ресурса документа отгрузки.
    /// </summary>
    public class UpdateResourceShipmentInput
    {       
        /// <summary>
        /// Id документа отгрузки.
        /// </summary>
        public int DocumentShipmentId { get; set; }

        /// <summary>
        /// Номер документа отгрузки.
        /// </summary>
        public string? DocumentShipmentNumberCode { get; set; }

        /// <summary>
        /// Дата документа отгрузки.
        /// </summary>
        public DateTime DocumentShipmentDate { get; set; }

        /// <summary>
        ///  Id клиента документа отгрузки.
        /// </summary>
        public int DocumentShipmentClientId { get; set; }

        /// <summary>
        /// Нужно ли подписать документ отгрузки.
        /// </summary>
        public bool IsSetActiveStatus { get; set; }

        /// <summary>
        /// Список входящих ресурсов в документ отгрузки.
        /// </summary>
        public ICollection<ModifyResourceShipmentInput>? ModifyResourceShipmentInputs { get; set; }
    }
}

File: warehouse-management-web-angular/src/app/warehouse/document-shipment/models/input/update-resource-shipment-input.ts
import { ModifyResourceShipmentInput } from './modify-resource-shipment-input';

/**
 * Класс входной модели редактирования ресурса документа отгрузки.
 */
export class UpdateResourceShipmentInput {
  /**
   * Id документа отгрузки.
   */
  documentShipmentId: number = 0;

  /**
   * Номер документа отгрузки.
   */
  documentShipmentNumberCode: string = '';

  /**
   * Дата документа отгрузки.
   */
  documentShipmentDate: Date = new Date();

  /**
   * Id клиента документа отгрузки.
   */
  documentShipmentClientId: number = 0;
  /**
   * Нужно ли подписать документ отгрузки.
   */
  isSetActiveStatus: boolean = false;

  /**
   * Список входящих ресурсов в документ отгрузки.
   */
  modifyResourceShipmentInputs: ModifyResourceShipmentInput[] = [];
}
